go/signedexchange/README.md

```diff
@@ -1,3 +1,83 @@
+## Overview
+We currently provide two command-line tools: `gen-signedexchange` and `gen-certurl`.
+
+`gen-signedexchange` generates a signed exchange file. The `gen-signedexchange` command constructs HTTP request and response pair from given command line flags, attach a cryptographic signature of the pair, and serializes the result to output file.
+
+`gen-certurl` converts a X.509 certificate chain to `application/cert-chain+cbor` format, which is defined in the [Section 3.3 of the Signed HTTP Exchanges spec](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#rfc.section.3.3).
+
+You are also welcome to use the code as golang lib (e.g. `import "github.com/WICG/webpackage/go/signedexchange"`), but please be aware that the API is not yet stable and is subject to change any time.
+
+## Getting Started
+
+### Prerequisite
+golang environment needs to be set up in prior to using the tool. We are testing the tool on latest golang. Please refer to [Go Getting Started documentation](https://golang.org/doc/install) for the details.
+
+### Installation
+We recommend using `go get` to install the command-line tool.
+
+```
+go get -u github.com/WICG/webpackage/go/signedexchange/cmd/...
+```
+
+### Creating our first signed exchange
+In this section, we guide you to create a signed exchange file, signed using self-signed certificate pair.
+
+Here, we assume that you have an access to a HTTPS server capable of serving static content. [1] Please substitute `https://yourcdn.example.net/` URLs to your web server URL.
+
+1. Prepare a file to be enclosed in the signed exchange. This serves as the content of the HTTP response in the signed exchange.
+    ```
+    echo "<h1>hi</h1>" > payload.html
+    ```
+
+1. Prepare a certificate and private key pair to use for signing the exchange. As of July 2018, we need to use self-signed certificate for testing, since there are no CA that issues certificate with ["CanSignHttpExchanges" extension](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#cross-origin-cert-req). To generate a signed exchange compatible self-signed key pair with OpenSSL, invoke:
+    ```
+    # Generate prime256v1 ecdsa private key.
+    openssl ecparam -out priv.key -name prime256v1 -genkey
+    # Create a certificate signing request for the private key.
+    openssl req -new -sha256 -key priv.key -out cert.csr \
+      -subj '/CN=example.org/O=Test/C=US'
+    # Self-sign the certificate with "CanSignHttpExchanges" extension.
+    openssl x509 -req -days 360 -in cert.csr -signkey priv.key -out cert.pem \
+      -extfile <(echo "1.3.6.1.4.1.11129.2.1.22 = ASN1:NULL\nsubjectAltName=DNS:example.org")
+    ```
+
+1. Convert the PEM certificate to `application/cert-chain+cbor` format using `gen-certurl` tool.
+    ```
+    # Fill in dummy data for OCSP/SCT, since the certificate is self-signed.
+    gen-certurl -pem cert.pem -ocsp <(echo ocsp) -sct <(echo sct) > cert.cbor
```

Comment:
@irori I'm not sure if this is good. Do you have any suggestion here?

Reply:
As long as you use Chrome with `--ignore-certificate-errors-spki-list`, this is fine.

`-sct` should be optional (if not, we should fix `gen-certurl`), so we can omit it.

For OCSP, we should be able to use `openssl ocsp` to set up an OCSP responder and generate an OCSP response, but it would complicate this instructions even further.

Reply:
Hmm. Leaving this as is for now. @irori would you mind updating here next week?